src/generation.rs

```diff
@@ -158,6 +282,114 @@ impl <'a> std::convert::From<&'a ConceptualRustType> for SerializingRustType<'a>
     }
 }
 
+/// Output code for deserialization. Includes meta information for better usage to prevent warnings.
+#[derive(Default)]
+struct DeserializationCode {
+    content: BlocksOrLines,
+    read_len_used: bool,
```

Comment:
what's the purpose of this flag?

Reply:
In the `DeserializeEmbeddedGroup` trait (only plain/basic groups use this) there is a `read_len` parameter which isn't always used e.g. when there are no fields that can vary in CBOR length depending on the data e.g. optional fields, or type choices where the choices have heterogenous CBOR lengths. We keep track of that to avoid an unused variable warning. This is from the core rustc warning, not just clippy. In the cases where it's not checked, it's always checked before the call in the regular `Deserialize` trait (of itself, or where it's embedded into another type) since its length will be checked straight away in a single static length check that checks all CBOR length that is static.